frontend/packages/schema-editor/src/components/SchemaTree/SchemaNode/ActionButtons/ActionButtons.tsx

```diff
@@ -32,12 +33,22 @@ export const ActionButtons = ({ pointer, className }: ActionButtonsProps) => {
           onClick={convertToReference}
         />
       )}
-      <ActionButton
-        color='danger'
-        icon={<TrashIcon />}
-        titleKey='general.delete'
-        onClick={deleteNode}
-      />
+      {isDefinition(node) ? (
+        <ActionButton
+          color='danger'
+          icon={<TrashIcon />}
+          titleKey='schema_editor.disable_deletion_info_for_used_definition'
+          onClick={deleteNode}
+          disabled={true}
+        />
+      ) : (
+        <ActionButton
+          color='danger'
+          icon={<TrashIcon />}
+          titleKey='general.delete'
+          onClick={deleteNode}
+        />
+      )}
     </div>
```

Comment:
I think this can be simplified to the following: 
```tsx
const actionButtonTitleKey = isDefinition(node) ? 'schema_editor.disable_deletion_info_for_used_definition' : 'general.delete';

<ActionButton
    color='danger'
    icon={<TrashIcon />}
    titleKey={actionButtonTitleKey} 
    onClick={deleteNode}
    disabled={isDefinition(node)}
/>
```